include/deal.II/base/work_stream.h

```diff
@@ -663,6 +663,163 @@ namespace WorkStream
 #  endif // DEAL_II_WITH_TBB
 
 
+
+#  ifdef DEAL_II_WITH_TASKFLOW
+    /**
+     * Mostly a copy of the 2nd implementation of the Workstream paper taking
+     * advantage of thread local lists for re-use. Uses taskflow for task
+     * scheduling rather than TBB. Currently does not support chunking.
+     */
+
+    namespace taskflow_no_coloring
+    {
+      template <typename Worker,
+                typename Copier,
+                typename Iterator,
+                typename ScratchData,
+                typename CopyData>
+
+      /**
+       *  The last two arguments in this function are for chunking support which
+       * currently does not exist but ideally will later. For now they are
+       * ignored but still here to permit existing programs to function
+       */
+      void
+      run(const Iterator                          &begin,
+          const typename identity<Iterator>::type &end,
+          Worker                                   worker,
+          Copier                                   copier,
+          const ScratchData                       &sample_scratch_data,
+          const CopyData                          &sample_copy_data,
+          const unsigned int queue_length = 2 * MultithreadInfo::n_threads(),
+          const unsigned int chunk_size   = 8)
```

Comment:
You'll get warnings about unused arguments if you don't un-name these arguments:
```suggestion
          const unsigned int /*queue_length*/ = 2 * MultithreadInfo::n_threads(),
          const unsigned int /*chunk_size*/   = 8)
```